R/count.R

```diff
@@ -805,3 +760,30 @@ rename_missing_values <- function(x) {
     }
   }, envir = x)
 }
+
+filter_numeric <- function(.data,
+                           numeric_cutoff,
+                           numeric_cutoff_stat,
+                           numeric_cutoff_column,
+                           treat_var,
+                           by = NULL) {
+
+  if (is.null(numeric_cutoff)) {
+    return(.data)
+  }
+
+    vals <- .data %>%
+      {if (is.null(numeric_cutoff_column)) . else filter(., !!treat_var == numeric_cutoff_column)} %>%
+      mutate(
+        pct = n/total,
+        distinct_pct = distinct_n/distinct_total
+      ) %>%
+      filter(!!sym(numeric_cutoff_stat) >= !!numeric_cutoff) %>%
```

Comment:
Thought - could we generalize this to just take a filter expression instead of separate parameters? You could just evaluate the parameter here. And to check compliance with the layer you could check the syntax like this:

```r
> all.vars(str2lang("x >= 1 & y <= 2"))
[1] "x" "y"
```

Reply:
@mstackhouse A question about how you think this would work for different numeric cutoffs. My first thought is we could add a flag if the logic should use percentages instead of counts. There are four different situations I can think of and this is how I would propose it works:
1. Counts (no column specified) ">= 10"
2. Counts (column specified) "Placebo >= 10"
3. Percentages (no column specified) "pct >= 5"
4. Percentages (columns specified) "Placebo:pct >= 5"
You could also string these together using & like you did above (Placebo:pct > 5 & >20)

Reply:
This just made me think - did you test this with additional cols variables? 

Also, what about uglier cases where the treatment group has spaces in the name? Would it have to be like ``Xan High:pct``?

But - I think that we should require they specify the variable. I think it makes it less ambiguous but also gives you less permutations to handle

Reply:
@mstackhouse For right now it either looks at one column, or all columns. Thinking more about this, I think the current implementation without using an expression would be a better user API. For the reasons you listed above, I think trying to parse out both the column name and the cutoff_stat would be difficult and error prone. I might be biased towards the solution because that's how I learned it but I also think its easier for a user to understand.d

Reply:
I think you're right. Would you be opposed to renaming the function though? I say this because "where" feels open ended, and I'd want to call it maybe `set_numeric_threshold()` instead - because we're not offering control over exactly how that "where" clause works.

Since this is experimental, we could then deprecate this and supercede it with `set_numeric_where()` if we have a better idea of exactly how we'd want to design it. 

Does this make sense? Let me know if I'm just being a pain.